src/main/java/com/jc/tm/database/dao/BaseDao.java

```diff
@@ -0,0 +1,23 @@
+package com.jc.tm.database.dao;
+
+import com.jc.tm.database.entity.Task;
+
+import java.util.List;
+
+public interface BaseDao<E> {
+
+    //create or insert
+    void insert(E e);
+
+    //update
+    void update(E e);
+
+    //get
+    E getById(int id);
+
+    //get all
+    List<E> getAll();
+
+    //delete
+    void delete(E e);
```

Comment:
everything ok, good job. but lets add return type for delete. it is usefull for future. if we will remove some object, but object not exists into database we can return null and according null we can recognize ttaht object doest nott exist and display correct error

Reply:
delete method was changed from void to boolean.